core/src/main/scala/chisel3/internal/firrtl/IR.scala

```diff
@@ -861,7 +862,28 @@ case class DefBlackBox(
   params: Map[String, Param])
     extends Component
 
-case class Circuit(name: String, components: Seq[Component], annotations: Seq[ChiselAnnotation], renames: RenameMap) {
-  def firrtlAnnotations: Iterable[Annotation] = annotations.flatMap(_.toFirrtl.update(renames))
+@data class Circuit(
+  name:       String,
+  components: Seq[Component],
+  @deprecated("Do not use annotations val of Circuit directly - use firrtlAnnotations instead. Will be removed in a future release",
+    "Chisel 3.5")
+  annotations: Seq[ChiselAnnotation],
+  renames:     RenameMap,
+  @deprecated("Do not use newAnnotations val of Circuit directly - use firrtlAnnotations instead. Will be removed in a future release",
+    "Chisel 3.5")
+  @since newAnnotations: Seq[ChiselToFirrtlAnnotations] = Seq.empty) {
+
+  def firrtlAnnotations: Iterable[Annotation] =
+    annotations.flatMap(_.toFirrtl.update(renames)) ++ newAnnotations.flatMap(
+      _.toFirrtlAnnotations.flatMap(_.update(renames))
+    )
+
+  def copy(
+    name:           String = name,
+    components:     Seq[Component] = components,
+    annotations:    Seq[ChiselAnnotation] = annotations,
+    renames:        RenameMap = renames,
+    newAnnotations: Seq[ChiselToFirrtlAnnotations] = newAnnotations
+  ) = Circuit(name, components, annotations, renames, newAnnotations)
```

Comment:
@jackkoenig  not sure if we need two different versions of copy (similar to apply)

Reply:
Also not sure if we need an implementation for "unapply"

Reply:
when I tried to overload copy - it gave some error from the data macro - just FYI - did not spend too much time debugging it - if we do need two different copy versions will spend time on that.

Reply:
I responded on the `Outdated` thread: https://github.com/chipsalliance/chisel3/pull/2628#discussion_r919492572